WordPress/Classes/Models/BasePost.m

```diff
@@ -181,7 +182,7 @@ - (void)setDateCreated:(NSDate *)localDate
      If the date is now or in the past, and the status is scheduled, set the status
      to published.
      */
-    if (self.date_created_gmt == nil) {
+    if (self.date_created_gmt == nil || [self.date_created_gmt isEqualToDate:self.dateModified]) {
```

Comment:
> If feels like these two conditions should be a new method in BasePost instead, in order to ease code maintenance.

I thought about it, but decided it wouldn't be proper for another reason.  `BasePost` should *only* contain properties and methods in common with `AbstractPost` and `ReaderPost`.  Right now its not doing a good job of that and contains things specific to `AbstractPost`.  
I'd rather see us migrate anything specific to `AbstractPost` into that class as a clean up task.